extensions/amp-form/0.1/amp-form.js

```diff
@@ -415,62 +427,69 @@ export class AmpForm {
       }
     });
 
-    this.form_.addEventListener(
-      'submit',
-      this.handleSubmitEvent_.bind(this),
-      true
-    );
-
-    this.form_.addEventListener(
-      'blur',
-      (e) => {
-        checkUserValidityAfterInteraction_(dev().assertElement(e.target));
-        this.validator_.onBlur(e);
-      },
-      true
-    );
-
-    this.form_.addEventListener(
-      AmpEvents.FORM_VALUE_CHANGE,
-      (e) => {
-        checkUserValidityAfterInteraction_(dev().assertElement(e.target));
-        this.validator_.onInput(e);
-      },
-      true
-    );
+    this.ampFormServicePromise_.then((ampFormService) => {
```

Comment:
Is there a type of extension service getter that is not async? (Since the extension is loaded anyway...)

Reply:
Is it guaranteed that the service loads before the AmpForm class?

Reply:
Seems like the sync version is preferred: 
`148:28  error  Forbidden: "getServiceForDoc". Synchronous access to element services is unreliable (#22414). Use getServicePromiseForDoc() instead  local/no-forbidden-terms`

See https://github.com/ampproject/amphtml/issues/22414

I feel that sync should be ok, but then again it could cause weird issues that are hard to repro/track down in the future, all for a small performance benefit. What are your thoughts?

Reply:
Discussed offline: we should be good.